csm_web/frontend/src/components/Section.js

```diff
@@ -172,6 +172,26 @@ class WeekAttendance extends React.Component {
 }
 
 class Attendances extends React.Component {
+  constructor(props) {
+    super(props);
+    this.shownWeeks = 2;
```

Comment:
Move shown weeks and button name into this.state

Directly assign state in constructor using `this.state = { ... }`

Use `this.setState` in all other methods to modify these properties. See the React tutorial for more details.